app/controllers/admin/settings/campaigns_controller.rb

```diff
@@ -0,0 +1,22 @@
+module Admin
+  module Settings
+    class CampaignsController < Admin::ApplicationController
+      def create
+        result = ::Campaigns::SettingsUpsert.call(settings_params)
+
+        if result.success?
+          Audit::Logger.log(:internal, current_user, params.dup)
+          redirect_to admin_config_path, notice: "Site configuration was successfully updated."
+        else
+          redirect_to admin_config_path, alert: "😭 #{result.errors.to_sentence}"
```

Comment:
[nonblocking] I've always done `errors.full_messages.to_sentence` so it took me a bit of reading through this PR to realize that this is returning an array of error messages instead of an `ActiveModel::Errors` instance.

I don't think it needs to change, just noting it here in case it becomes a common point of confusion.

Reply:
Yes. Generally, we use our custom wrapper (`ApplicationRecord#errors_as_sentence`) which does exactly what you said. These service objects — which so far I haven't refactored but kept consistent to the pre-existing one — return an array already so it's a bit different here.